### ISSUE TITLE
Install Docker Desktop 2.2.03 failed on Window10 Pro

### ISSUE THREAD

**Issue by cycbluesky** [docker/for-win#6004]
I uninstalled a pre-version docker desktop and download 2.2.03 to install failed.
My machine is 
OS Name	Microsoft Windows 10 Pro
Version	10.0.18362 Build 18362
Next is error log.
------------------------------------------------------------------
The flat name of the target computer could not be retrieved (2138).
   at System.DirectoryServices.AccountManagement.SAMStoreCtx.LoadComputerInfo()
   at System.DirectoryServices.AccountManagement.SAMStoreCtx.get_MachineUserSuppliedName()
   at System.DirectoryServices.AccountManagement.SAMStoreCtx.FindNativeByNT4IdentRef(Type principalType, String urnValue)
   at System.DirectoryServices.AccountManagement.SAMStoreCtx.FindPrincipalByIdentRef(Type principalType, String urnScheme, String urnValue, DateTime referenceDate)
   at System.DirectoryServices.AccountManagement.Principal.FindByIdentityWithTypeHelper(PrincipalContext context, Type principalType, Nullable`1 identityType, String identityValue, DateTime refDate)
   at System.DirectoryServices.AccountManagement.Principal.FindByIdentityWithType(PrincipalContext context, Type principalType, String identityValue)
   at System.DirectoryServices.AccountManagement.GroupPrincipal.FindByIdentity(PrincipalContext context, String identityValue)
   at CommunityInstaller.CreateGroupAction.<DoAsync>b__33_0()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at CommunityInstaller.InstallWorkflow.<HandleD4WPackageAsync>d__29.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<ProcessAsync>d__24.MoveNext()

**Comment by SevnChen** [comment]
I had the same problem, did you solve it?

**Comment by SevnChen** [comment]
It's strange. I can install Docker Desktop 2.0.0.0-win81 without any problems

**Comment by Poliwrath** [comment]
Enable / start the "Workstation" service

**Comment by cycbluesky** [comment]
@Poliwrath Thanks! I found the Workstation service crashed due to windows update failed. After I upgaded windows to version 1909, the Workstation service was back. And now I installed Docker Desktop 2.2.03 successfully.
@SevnChen try to check your Workstation service.

**Comment by ivanov** [comment]
I got the same error after Docker prompted to update itself. Not only did the update fail, but it also deleted the previous installation so I ended up with Docker after this and had to re-download all 900Mb of Docker for Windows.exe.

<details>
```
The flat name of the target computer could not be retrieved (2138).
   at System.DirectoryServices.AccountManagement.SAMStoreCtx.LoadComputerInfo()
   at System.DirectoryServices.AccountManagement.SAMStoreCtx.get_MachineUserSuppliedName()
   at System.DirectoryServices.AccountManagement.SAMStoreCtx.FindNativeByNT4IdentRef(Type principalType, String urnValue)
   at System.DirectoryServices.AccountManagement.SAMStoreCtx.FindPrincipalByIdentRef(Type principalType, String urnScheme, String urnValue, DateTime referenceDate)
   at System.DirectoryServices.AccountManagement.Principal.FindByIdentityWithTypeHelper(PrincipalContext context, Type principalType, Nullable`1 identityType, String identityValue, DateTime refDate)
   at System.DirectoryServices.AccountManagement.Principal.FindByIdentityWithType(PrincipalContext context, Type principalType, String identityValue)
   at System.DirectoryServices.AccountManagement.GroupPrincipal.FindByIdentity(PrincipalContext context, String identityValue)
   at CommunityInstaller.CreateGroupAction.<DoAsync>b__33_0()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at CommunityInstaller.InstallWorkflow.<HandleD4WPackageAsync>d__29.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<ProcessAsync>d__24.MoveNext()
```
</details>

**Comment by mat007** [comment]
Thanks for the reports and sorry for the inconvenience.

It probably is because for some reason you don’t have the Workstation system service started.
We have added a prerequisite check in the installer to try to start it in that case and if that fails report an error (at least it won’t start the installation process and uninstall the previous version if any).
It’s been released to the Edge branch starting with 2.2.3.0, but not yet to the Stable branch (soon!).

If you’re still having this problem after updating to or past to 2.2.3.0 please shout!

**Comment by ivanov** [comment]
I forgot to come back here and report that starting the Workstation service did also fix this issue for me.
